easy_my_coop/models/partner.py

```diff
@@ -84,11 +85,11 @@ def _get_share_type(self):
                  'share_ids.share_number')
     def _compute_cooperator_type(self):
         for partner in self:
-            share_type = ''
-            for line in partner.share_ids:
-                share_type = line.share_product_id.default_code
-            if share_type:
-                partner.cooperator_type = share_type
+            if partner.share_ids and partner.share_ids[0].share_number > 0:
+                share = partner.share_ids[0]
```

Comment:
I understand that you want to.
- not loop over all lines if not necessary
- check if they actually have shares of that type (i.e. `share_number > 0`)

But in what you propose here, you only check for the first share id `share_ids[0]`. If there are no shares for that one, `''` will be returned, but there could be shares for the `share_ids[1]` (since the `share.lines` are not ordered on e.g. share_number - that could also be a way to fix it). 

So I think a loop is necessary. You could stop it as soon as a value is found though.

Reply:
I think Manu is right on that one.

On a side note, it'd be cleaner to have a commit message / PR title that actually says what's being changed. eg: `[REF] emc: get share type from first share of the partner`

Reply:
@mclaeysb i've taken your comment into account. could review it again ?

Reply:
Cool!